### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by romulogcerqueira**
* Cleared Code;
* Created some dynamic ports (range, shader_viewer, start_bearing...);
* Changed updateMultibeamSonarPose() and updateScanningSonarPose().

### CODE REVIEW

File: imaging_sonar_simulation.orogen [hunk 1]
```diff
@@ -17,12 +18,20 @@ task_context "Task" do
     # set the sonar pose using a RigidBodyState structure
     input_port("sonar_pose_cmd", "base::samples::RigidBodyState");
 
+    # New Sonar Viewer
+    output_port("sonar_viewer", "base::samples::frame::Frame");
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Use ro_ptr("/base/samples/frame/Frame") - you can have a look at underwater_camera_simulation to see how it's done. It is a smart pointer, thus avoiding a lot of unnecessary copying around.

File: imaging_sonar_simulation.orogen [hunk 2]
```diff
@@ -17,12 +18,20 @@ task_context "Task" do
...
+
+	# Shader Viewer
+    output_port("shader_viewer", "base::samples::frame::Frame");
+
     # Range of the sonar
     property("range", "double", 50.0).dynamic
-
+    
+    # Gain of the sonar
+    property("gain", "int", 50).dynamic
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Use a float gain

File: imaging_sonar_simulation.orogen [hunk 3]
```diff
@@ -54,7 +62,13 @@ task_context "MultibeamSonarTask" do
...
     # Number of Beams
     property("number_of_beams", "int", 256).dynamic
-end
\ No newline at end of file
+    
+    # Angular Resolution
+    property("angular_resolution", "double", 1.0).dynamic
+    
+    # Start Bearing
+    property("start_bearing", "double", -60.0).dynamic
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Every value on an orogen component interface MUST use SI units (in this case, radians)

File: imaging_sonar_simulation.orogen (hunk 1 above)
Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: Done!

File: imaging_sonar_simulation.orogen (hunk 2 above)
Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: Done!

File: imaging_sonar_simulation.orogen (hunk 3 above)
Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: I agreed with you. Done!

File: tasks/MultibeamSonarTask.cpp [hunk 4]
```diff
@@ -55,59 +71,63 @@ bool MultibeamSonarTask::startHook() {
...
-	std::vector<cv::Mat> raw_intensity;
-	std::vector<uint8_t> sonar_data;
+	if (_sonar_pose_cmd.read(linkPose) == RTT::NewData) {
 
-	for (int i = 0; i < _multibeam_sonar.getNumberOfBeams(); i++) {
-		raw_intensity.push_back(_multibeam_sonar.decodeShaderImage(cv_beams[i]));
-		std::vector<uint8_t> current_data = _multibeam_sonar.getPingData(raw_intensity[i]);
-		sonar_data.insert(sonar_data.end(), current_data.begin(), current_data.end());
+		base::samples::RigidBodyState multibeamSonarPose = linkPose;
+		updateMultibeamSonarPose(multibeamSonarPose);
```
Comment (doudou) [inline tasks/MultibeamSonarTask.cpp]: Why not `updateMultiBeamSonarPose(linkPose)` directly ?

File: tasks/MultibeamSonarTask.cpp [hunk 5]
```diff
@@ -55,59 +71,63 @@ bool MultibeamSonarTask::startHook() {
...
+	MultibeamSonarTaskBase::updateHook();
 
-	// split shader image to beams
-	std::vector<cv::Mat> cv_beams = _multibeam_sonar.splitShaderImage(cv_image);
+	base::samples::RigidBodyState linkPose;
 
-	// decode shader image and get ping data
-	std::vector<cv::Mat> raw_intensity;
-	std::vector<uint8_t> sonar_data;
+	if (_sonar_pose_cmd.read(linkPose) == RTT::NewData) {
```
Comment (doudou) [inline tasks/MultibeamSonarTask.cpp]: This should be a while() not an if()

File: imaging_sonar_simulation.orogen [hunk 6]
```diff
@@ -40,11 +49,10 @@ task_context "ScanningSonarTask" do
     property("start_angle", "double", 0.0).dynamic
 
     # Right limit angle
-    property("end_angle", "double", 360.0).dynamic
+    property("end_angle", "double", 6.28318531).dynamic
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: You can use `2 * Math::PI` here

File: imaging_sonar_simulation.orogen [hunk 7]
```diff
@@ -40,11 +49,10 @@ task_context "ScanningSonarTask" do
...
 
     # Step angle size
-    property("step_angle", "double", 1.8).dynamic
-
+    property("step_angle", "double", 0.0314159265).dynamic
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Same, `Math::PI / 180 * 1.8`

File: tasks/MultibeamSonarTask.cpp (hunk 4 above)
Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: You are right! :D

Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: Done!

File: tasks/MultibeamSonarTask.cpp (hunk 5 above)
Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: Done!

File: imaging_sonar_simulation.orogen (hunk 6 above)
Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: I didn't know about "Math::PI" in orogen component interface. It a good tool! Thanks!

Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: Done!

File: imaging_sonar_simulation.orogen (hunk 7 above)
Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: Done!

**Comment by romulogcerqueira** [comment]
ping @doudou

File: imaging_sonar_simulation.orogen [hunk 8]
```diff
@@ -17,12 +18,20 @@ task_context "Task" do
...
+
+	# Shader Viewer
+    output_port("shader_viewer", ro_ptr("/base/samples/frame/Frame"));
+
     # Range of the sonar
     property("range", "double", 50.0).dynamic
-
+    
+    # Gain of the sonar
+    property("gain", "double", 75.0).dynamic
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Gain of 75 ? Isn't that ... a lot ?

Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: It is just the default value. We can adjust during the simulation. What do you suggest me? 50, 60%?

Comment (doudou) [inline imaging_sonar_simulation.orogen]: > 50, 60%?

AAaaaaah

Same comment than with the radians. There's no such thing as a percentage. Use 0.75

Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: Done and updated!

**Pull request merged**